packages/deployments/contracts/deploy/deploy.ts

```diff
@@ -177,6 +207,7 @@ const func: DeployFunction = async (hre: HardhatRuntimeEnvironment): Promise<voi
       from: deployer.address,
       args: [WRAPPED_ETH_MAP.get(+chainId)],
       log: true,
+      skipIfAlreadyDeployed: true,
```

Comment:
Do we know if this matches the compiled bytecode against what's onchain? Seems like it should, but just curious.

If it does, and it works for proxies, we should do this with things like the `BridgeRouter`, `TokenRegistry`, etc.

Reply:
`const deployedImpl = await hre.upgrades.prepareUpgrade(beaconAddress, factory);`

prepareUpgrade will be the solution. finally fixed.
So if the implementation wasn't changed, deploying script won't redeploy implementation and call `upgradeTo` as well.

Great plugin if updated a bit!